### PR TITLE
Refactor utils::format_timespan to use chrono types

### PR DESCRIPTION
**Pull request by Vultraz**
Required a rather in-depth refactor since we're now dealing with distinct types for the different durations. This also adjusts the periods for years and months. Previously, we were using values of 30 days for a month and 12 months for year. Now, we use the chrono values of a month as 1/12 of a year and a year as 365.2425 days.

Requesting a review from Iris since she implemented the function originally.

### CODE REVIEW

File: src/serialization/chrono.hpp [hunk 1]
```diff
@@ -72,15 +72,29 @@ inline auto parse_duration(const config_attribute_value& val, const Duration& de
 	return Duration{val.to_long_long(def.count())};
 }
 
-template<typename Rep, typename Period>
-constexpr auto deconstruct_duration(const std::chrono::duration<Rep, Period>& span)
+template<typename... Ts, typename Rep, typename Period>
+constexpr auto deconstruct_duration(const std::tuple<Ts...>&, const std::chrono::duration<Rep, Period>& span)
```
Comment (soliton-) [inline src/serialization/chrono.hpp:76]: What's the point of the tuple argument? IMO calls would be more readable without it.

Comment (Vultraz) [reply src/serialization/chrono.hpp:76]: Allows the compiler to deduce the parameter pack by passing in a pre-set tuple. I figured it was cleaner than having to specify the template parameters every time, but I'm opening to refining the design.

**Comment by irydacea** [comment]
Removed myself because it's been too long and I don't remember what the code did or what it was used for.

File: src/serialization/chrono.hpp (hunk 1 above)
Comment (soliton-) [reply src/serialization/chrono.hpp:76]: If the implication is that shorter is cleaner then I disagree. I don't see how
`auto [days, hours, minutes, seconds] = chrono::deconstruct_duration(chrono::format::days_hours_mins_secs, time_up)`
looks cleaner than
`auto [days, hours, minutes, seconds] = chrono::deconstruct_duration<chrono::days, std::chrono::hours, std::chrono::minutes, std::chrono::seconds>(time_up)`.
For the former besides looking up `deconstruct_duration` I also need to look up whatever `chrono::format::days_hours_mins_secs` is.

**Pull request merged**